src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAICrafting.java

```diff
@@ -189,6 +194,29 @@ protected IAIState getNextCraftingState()
         return GET_RECIPE;
     }
 
+    /**
+     * Check if stack is part of the recipe.
+     * @param stack the stack to check.
+     * @param currentRecipeStorage the recipe to compare.
+     * @return true if so.
+     */
+    private boolean isPartOfRecipe(final ItemStack stack, final IRecipeStorage currentRecipeStorage)
+    {
+        if (ItemStackUtils.compareItemStacksIgnoreStackSize(stack, currentRecipeStorage.getPrimaryOutput()))
+        {
+            return true;
+        }
+
```

Comment:
What about required tools?  Or tools not required for this particular recipe but required for some other part of their job?  This seems like it might get triggered more often than perhaps is intended.